java/arcs/sdk/storage/StorageKey.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.sdk.storage
+
+/**
+ * Represents a location in Arcs' storage system where an [Entity], [Collection], [Singleton], or a
+ * region of any of those items can be found.
+ */
+typealias StorageKey = arcs.core.storage.StorageKey
```

Comment:
We use type aliases to expose things publicly, are these types we intend to be used by particle authors or folks calling high level arcs control APIs like startArc?
If we're unsure, or if this is temporary while the prodex work advances, I'd rather us have the helpers we write for fedex/people take deps on core temporarily.

Reply:
I think I like Maria's suggestion to typealias the sdk-level's StorageKey to String, and then rely on the StorageKey impls' parsers to construct the real thing.

Reply:
Either a typealias or an `inline class`

Reply:
Why do Particle developers need to specify StorageKeys? Letting the particle dynamically control and specify them after they've been pre-compiled as a plan/allocator, seems to me to constitute an surface area for exploit.